src/util/audio/VorbisProducer.cpp

```diff
@@ -36,23 +38,33 @@ bool VorbisProducer::start(std::string filename, unsigned int timestamp)
 			{
 				size_t numFrames = 1;
 				auto sampleBuffer = new float[1024 * this->sfInfo.channels];
-				std::unique_lock<std::mutex> lock(audioQueue->syncMutex());
-
-				while (!this->stopProducer && numFrames > 0 && !this->audioQueue->hasStreamEnded())
```

Comment:
You acidentially took the wrong choice in your merge here. You need both the lock and should use `numFrames`.